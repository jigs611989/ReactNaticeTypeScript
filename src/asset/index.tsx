export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
