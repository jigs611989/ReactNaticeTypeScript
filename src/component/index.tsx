export { default as Button } from './Button';
export { default as Label } from './Label';
export { default as UserName } from './UserName';
export { default as TextField } from './TextField';
export { default as SliderButton } from './SliderButton';
