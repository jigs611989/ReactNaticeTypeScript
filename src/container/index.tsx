export { default as WelcomeScreen } from './WelcomeScreen';
export { default as HomeScreen } from './HomeScreen';
export { default as ButtonOptionScreen } from './ButtonOptionScreen';
