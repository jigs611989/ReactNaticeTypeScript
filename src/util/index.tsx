import { regex } from '../constant/AppConstant';
const isValidName = (name: string): boolean => {
  if (name.match(regex.name)) {
    return true;
  }
  return false;
};

export { isValidName };
