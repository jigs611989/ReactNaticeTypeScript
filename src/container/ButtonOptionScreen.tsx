import React from 'react';
import { Alert, SafeAreaView, View } from 'react-native';
import { String } from '../asset';
import { Button, Label, SliderButton, UserName } from '../component';
import style from './style/ButtonOptionScreen';

const ButtonOptionScreen: React.FC = () => {
  return (
    <SafeAreaView style={[style.rootContainer, style.container]}>
      <UserName />
      <View style={style.spacer} />
      <Label title={String.buttonVariationTitle} style={style.labelStyle} />
      <Button
        testID={'ButtonOption1'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption1)}
        textStyle={style.buttonOption1}
      />
      <Button
        testID={'ButtonOption2'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption2)}
        style={style.buttonOption2}
      />
      <Button
        testID={'ButtonOption3'}
        title={String.Buttons.pressMe}
        onPress={() => Alert.alert('', String.Alert.buttonOption3)}
        style={style.buttonStyle}
      />
      <SliderButton
        testID={'ButtonOption4'}
        onPress={() => Alert.alert('', String.Alert.buttonOption4)}
      />
    </SafeAreaView>
  );
};

export default ButtonOptionScreen;
