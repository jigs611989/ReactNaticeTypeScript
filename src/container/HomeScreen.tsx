import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { SafeAreaView, Image } from 'react-native';
import { String, Images } from '../asset';
import { Button, UserName } from '../component';
import { Screens } from '../constant/AppConstant';
import style from './style/HomeScreen';

const HomeScreen: React.FC = () => {
  const { navigate } = useNavigation();
  return (
    <SafeAreaView style={[style.rootContainer, style.container]}>
      <UserName />
      <Image source={Images.reactLogo} style={style.image} />
      <Button
        title={String.Buttons.buttonOption}
        onPress={() => navigate(Screens.ButtonOptionScreen)}
        style={style.buttonStyle}
      />
    </SafeAreaView>
  );
};

export default HomeScreen;
