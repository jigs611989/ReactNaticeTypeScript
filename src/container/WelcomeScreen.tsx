import { useNavigation } from '@react-navigation/native';
import React, { useCallback, useState, useEffect, useRef } from 'react';
import { Alert, SafeAreaView, View } from 'react-native';
import { useDispatch } from 'react-redux';
import Strings from '../asset/String';
import { Button, Label, TextField, PopUp } from '../component';
import { Screens } from '../constant/AppConstant';
import { saveName } from '../redux/action/name';
import { isValidName } from '../util';
import style from './style/WelcomeScreen';
import { checkDeviceType } from '../nativeModule/DeviceType';

const WelcomeScreen: React.FC = () => {
  const popupRef = useRef();
  const { navigate } = useNavigation();
  const [name, setName] = useState('');
  const [deviceType, setDeviceType] = useState('');
  const dispatch = useDispatch();

  useEffect(() => {
    checkDeviceType((error, type) => {
      // Alert.alert('You are running on ' + type);
      setDeviceType(type)
      popupRef.current?.show()
    });
  }, []);

  const onPressContinue = useCallback(() => {
    if(isValidName(name)) {
      dispatch(saveName(name))
      navigate(Screens.AppStack)
    } else {
      Alert.alert('', Strings.Alert.invalidName)
    }
  }, [name])

  return (
    <SafeAreaView style={[style.rootContainer, style.container]}>
      <View style={style.spacer}/>
      <Label title={Strings.welcome} />
      <View style={style.spacer}/>
      <TextField 
        placeholder={Strings.enterYourName}
        value={name}
        onChangeText={(text) => setName(text)}
        autoCorrect={false}
      />
      <Button 
        title={Strings.Buttons.continue} 
        onPress={onPressContinue}
        style={style.buttonStyle}
      />
      <View style={style.spacer}/>
      <PopUp
        ref={popupRef}
        title={'Device Type'}
        description={`You are running on ${deviceType}`}
        onPress={() => popupRef.current?.hide()}
      />
    </SafeAreaView>
  );
};

export default WelcomeScreen;
