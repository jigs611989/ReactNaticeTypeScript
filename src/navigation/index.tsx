import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import HomeScreen from '../container/HomeScreen';
import WelcomeScreen from '../container/WelcomeScreen';
import ButtonOptionScreen from '../container/ButtonOptionScreen';
import Screens from '../constant/Screens';

const Stack = createStackNavigator();

function OnboardingStack() {
  return (
    <Stack.Navigator headerMode="none">
      <Stack.Screen name={Screens.WelcomeScreen} component={WelcomeScreen} />
    </Stack.Navigator>
  );
}

function AppStack() {
  return (
    <Stack.Navigator>
      <Stack.Screen name={Screens.HomeScreen} component={HomeScreen} />
      <Stack.Screen
        name={Screens.ButtonOptionScreen}
        component={ButtonOptionScreen}
      />
    </Stack.Navigator>
  );
}

function MainNavigation() {
  return (
    <NavigationContainer>
      <Stack.Navigator headerMode={'none'}>
        <Stack.Screen
          name={Screens.OnboardingStack}
          component={OnboardingStack}
        />
        <Stack.Screen name={Screens.AppStack} component={AppStack} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default MainNavigation;
