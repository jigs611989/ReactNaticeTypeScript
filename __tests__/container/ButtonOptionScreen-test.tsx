import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { ButtonOptionScreen } from '../../src/container';
import { render, fireEvent } from '@testing-library/react-native';
import { Alert } from 'react-native';
import Slider from 'react-native-slide-to-unlock';

const mockStore = configureStore();

it('renders correctly', () => {
  const initialState = { name: '' };
  const store = mockStore(initialState);
  const { toJSON, getByTestId, UNSAFE_getByType } = render(
    <Provider store={store}>
      <ButtonOptionScreen />
    </Provider>,
  );

  const buttonOption1 = getByTestId('ButtonOption1');
  fireEvent(buttonOption1, 'onPress');
  const buttonOption2 = getByTestId('ButtonOption2');
  fireEvent(buttonOption2, 'onPress');
  const buttonOption3 = getByTestId('ButtonOption3');
  fireEvent(buttonOption3, 'onPress');
  const buttonOption4 = UNSAFE_getByType(Slider);
  fireEvent(buttonOption4, 'onEndReached');

  expect(Alert.alert).toHaveBeenCalledTimes(4);
  expect(toJSON()).toMatchSnapshot();
});
