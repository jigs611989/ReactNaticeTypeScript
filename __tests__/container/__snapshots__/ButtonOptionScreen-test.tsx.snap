// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#0B121E",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "lineHeight": 28,
          "margin": 16,
          "marginBottom": 2,
        }
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "flex": 1,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "lineHeight": 28,
          "margin": 16,
          "marginBottom": 2,
        },
        Object {
          "color": "#adff2f",
        },
      ]
    }
  >
    4 variations of a button
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="ButtonOption1"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          Object {
            "color": "blue",
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#008080",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="ButtonOption2"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f7a934",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="ButtonOption3"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View />
</RCTSafeAreaView>
`;
